### PR TITLE
feat: allow to pass promise as response

### PR DESCRIPTION
**Pull request by maxime1992**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Build Lib
         run: yarn build
       - name: Release
-        if: contains('refs/heads/master', github.ref)
+        if: contains('refs/heads/master refs/heads/feat-allow-to-pass-promise-as-response', github.ref)
```
Comment (maxime1992) [inline .github/workflows/main.yml:28]: remove this

File: .releaserc [hunk 2]
```diff
@@ -1,6 +1,11 @@
 {
   "pkgRoot": "dist",
   "branches": [
-    "master"
+    "master",
+    {
+      "name": "feat-allow-to-pass-promise-as-response",
+      "channel": "feat-allow-to-pass-promise-as-response",
+      "prerelease": true
+    }
```
Comment (maxime1992) [inline .releaserc:9]: remove this

**Comment by maxime1992** [comment]
@zakhenry I've made a pre-release to test in my current branch and this works really well. It's letting me delay a request and in the e2e test make sure that a loading page/spinner is displayed and then resolve that promise as soon as the check is done to let the grpc call finish.

File: src/playwright/mock-grpc-unary.ts [hunk 3]
```diff
@@ -6,20 +6,24 @@ import { readGrpcRequest } from './read-grpc-request';
...
   const url = `/${rpc.service.serviceName}/${rpc.methodName}`;
 
   // note this wildcard route url base is done in order to match both localhost and deployed service usages.
-  await (mockAtContextLevel ? page.context() : page).route('**' + url, (route) => {
+  await (mockAtContextLevel ? page.context() : page).route('**' + url, async (route) => {
     expect(route.request().method(), 'ALL gRPC requests should be a POST request').toBe('POST');
 
-    const grpcResponse = typeof response === 'function' ? response(readGrpcRequest(route.request())) : response;
+    const grpcResponseWrapped = typeof response === 'function' ? response(readGrpcRequest(route.request())) : response;
+    const grpcResponse = await Promise.resolve(grpcResponseWrapped);
```
Comment (zakhenry) [inline src/playwright/mock-grpc-unary.ts:22]: minor possible cleanup

`const grpcResponse = await (typeof response === 'function' ? response(readGrpcRequest(route.request())) : response )`

but I think they end up functionally equivalent anyway though one less call on the stack

**Review by zakhenry — APPROVED** [review]

**Pull request merged**